### PR TITLE
Fix missing sanitizing of subpackage targetTypes

### PR DESCRIPTION
**Pull request by kinke**
A regression introduced in #2277.

### CODE REVIEW

File: source/dub/recipe/io.d [hunk 1]
```diff
@@ -132,6 +140,23 @@ unittest { // issue #711 - configuration default target type not correct for SDL
 	}
 }
 
+unittest { // make sure targetType of sub packages are sanitized too
+	import dub.compilers.buildsettings : TargetType;
+	auto inputs = [
+		"dub.sdl": "name \"test\"\nsubPackage {\nname \"sub\"\ntargetType \"sourceLibrary\"\nconfiguration \"a\" {\n}\n}",
+		"dub.json": "{\"name\": \"test\", \"subPackages\": [ { \"name\": \"sub\", \"targetType\": \"sourceLibrary\", \"configurations\": [{\"name\": \"a\"}] } ] }"
```
Comment (Geod24) [inline source/dub/recipe/io.d:147]: You might want to use backtip-delimited strings here ?

Comment (kinke) [reply source/dub/recipe/io.d:147]: I just copy&pasted it from the other unittests above. Backticks would be usable for the JSON, yes, but not for the sdl (in a single line), so for consistency...

**Comment by kinke** [comment]
> Well, thinking about this more, this goes against what subpackages are meant to be

? - Each subpackage recipe is sanitized separately, not inheriting anything from the parent package if that's what you meant.

**Review by Geod24 — APPROVED** [review]
Thanks for tracking this. Failure looks spurious, retried the CI.

**Pull request merged**